cicd/_common_deploy_logic.sh

```diff
@@ -112,17 +123,19 @@ function transform_arg {
     # transform components to "$1" options for bonfire
     options=""
     option="$1"; shift;
-    components="$@"
+    components="$*"
```

Comment:
this can be a big change, along with the change below I'd be interested if any component behavior changes.

Reply:
This line was violating [SC2124](https://www.shellcheck.net/wiki/SC2124), "Assigning an array to a string" 

Instead of keeping the array, I thought it was simpler to remove the need of an array there and use a single string "$*" for the whole list of parameters , and then make use of the split by spaces without quoting the components list variable ( see lines 134  and 139 of the same script ) . 

@Ichimonji10  may have an opinion here too :)

Reply:
Sure, as long as it's accounted for. I was just highlighting that this is not normally a drop in replacement for `$@`.

Reply:
It sounds like you've got a handle on this already. Might any of the parameters ever contain whitespace? If so, this is problematic; if not, this is OK.

I don't really have time to read through this entire script and understand it, but this looks like it either normalizes or extracts parameters in `$@`. If so, an alternate approach is to use `getopt` (not `getopts`); see [getopt(1)](https://man.archlinux.org/man/core/util-linux/getopt.1.en), `/usr/share/doc/util-linux/`, or [this](https://github.com/pulp/pulp-fixtures/blob/f319a1831702e871605d82f260eb4560a6483f79/file/gen-fixtures.sh#L33-L61).

Reply:
Yep - exactly what you said @Ichimonji10  - and nope, no spaces expected (they're supposed to be component identifiers without spaces on them so we should be good. 

About `get-opt` the thing is the parameters are read from a variable and not from a CLI option - getopt seems a bit over-engineered IMO, to basically process a list of space separated component ids, which is essentially all that's happening here. 

I appreciate your time with the review!